### PR TITLE
fix: preserve verbatim request id bytes through response

### PR DESCRIPTION
**Pull request by kasrakhosravi**
## Summary

Today the request id is parsed into `interface{}` (Go's JSON decoder represents numbers as `float64`) and then cast to `int64`. Two consequences:

- Integers above `2^53` lose precision (nanosecond timestamps used by some indexers, big counters, etc.).
- Fractional ids get silently truncated (`3.14` → `3`).

- Request `id: 9007199254740993` (= 2⁵³+1) → response `id: 9007199254740992`.
- Request `id: 3.14` → response `id: 3`.

## Approach

- Capture the verbatim id bytes during `JsonRpcRequest.UnmarshalJSON` into a new `idRaw` field.
- Add `IDRawBytes()` getter.
- Add `JsonRpcResponse.SetIDBytesPreserving(b []byte)` that stores raw bytes without canonicalizing through a `float64→int64` round-trip.
- `Network.normalizeResponse` prefers the raw id bytes when available, falls back to the existing typed-id path for programmatically-built requests.

## Backward compatibility

- Small ints and strings round-trip byte-for-byte identically (verified with unit tests).
- No code reads `r.ID.(int64)` / `.(float64)` — only pass-through usage exists today, so the typed view is preserved as-is.
- Pure superset of current behavior: ids that *would have been* corrupted are now preserved; everything else is unchanged.

## Test plan

- [x] `TestNormalizeResponse_IDByteFidelity` — 6 sub-cases covering small int, zero, string, large int (>2⁵³), fractional, near-uint64-max
- [x] `TestJsonRpcRequest_IDRawBytes` — 7 sub-cases verifying `IDRawBytes()` for each id shape, including null id, missing id, and programmatic construction
- [x] `go test ./common/ ./architecture/evm/ ./erpc/ -run TestNormalize` all green

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Medium Risk**
> Changes JSON-RPC request/response ID handling to prefer verbatim wire bytes, which affects all response normalization paths and could surface edge cases with unusual `id` formats; coverage is improved via new fidelity tests.
> 
> **Overview**
> Ensures JSON-RPC responses echo the client’s original `id` **byte-for-byte** (including integers >2^53, fractional IDs, and non-canonical numeric formats) instead of re-marshalling through a lossy typed representation.
> 
> This captures raw request `id` bytes during `JsonRpcRequest.UnmarshalJSON`, propagates them through `Clone()`, clears them on `SetID`, and updates `Network.normalizeResponse` to prefer `IDRawBytes()` and set the response via `JsonRpcResponse.SetIDBytes` (which no longer canonicalizes `idBytes`). Adds unit tests pinning end-to-end ID fidelity and the new raw-id contracts.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 5cdf62edd26e36d6048d093eb76c56a698e4a576. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: common/json_rpc.go [hunk 1]
```diff
@@ -260,6 +260,33 @@ func (r *JsonRpcResponse) SetIDBytes(idBytes []byte) error {
 	return r.parseID()
 }
 
+// SetIDBytesPreserving stores the response id from raw JSON bytes verbatim,
+// without canonicalizing them through a float64→int64 round-trip. The wire
+// output uses idBytes directly, so this preserves byte-for-byte fidelity for
+// large integers (>2^53), fractional ids, and exotic numeric formats. The
+// parsed r.id is populated best-effort for callers that read the typed view;
+// precision loss there is acceptable because the wire output uses idBytes.
+func (r *JsonRpcResponse) SetIDBytesPreserving(idBytes []byte) error {
```
Comment (aramalipoor) [inline common/json_rpc.go]: Why don't we fix/improve SetIDBytes itself?

**Review by aramalipoor — APPROVED** [review]
